### PR TITLE
Create sample for Idempotent Receiver

### LINKED ISSUE #76: Create sample for Idempotent Receiver
Create sample for [Idempotent Receiver](https://www.enterpriseintegrationpatterns.com/patterns/messaging/IdempotentReceiver.html).

### PR DESCRIPTION
**Pull request by ShammiL**
Fixes #76

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: idempotent-receiver/idempotent-receiver.bal [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+map<OrderStatus> orderStatuses = {};
+
+service /api/v1 on new http:Listener(8080) {
+
+    resource function put manage\-orders(OrderDetail orderDetail) returns
+        http:STATUS_ACCEPTED|http:STATUS_ALREADY_REPORTED {
+        OrderStatus? orderStatus = orderStatuses[orderDetail.orderId];
+        if orderStatus != null && orderStatus == orderDetail.status {
```
Comment (LakshanWeerasinghe) [inline idempotent-receiver/idempotent-receiver.bal]:
```suggestion
        if orderStatus != () && orderStatus == orderDetail.status {
```

Comment (LakshanWeerasinghe) [reply idempotent-receiver/idempotent-receiver.bal]: Isn't `orderStatus == orderDetail.status` check if enough?

File: idempotent-receiver/idempotent-receiver.bal [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    SHIPPED,
+    COMPLETED,
+    CANCELLED
+};
+
+map<OrderStatus> orderStatuses = {};
+
+service /api/v1 on new http:Listener(8080) {
+
+    resource function put manage\-orders(OrderDetail orderDetail) returns
```
Comment (LakshanWeerasinghe) [inline idempotent-receiver/idempotent-receiver.bal]:
```
resource function put manage\-orders\[string orderId](OrderDetail orderDetail) returns
```

Or
```
resource function put orders\[string orderId]\status(OrderDetail orderDetail) returns
```

Comment (LakshanWeerasinghe) [reply idempotent-receiver/idempotent-receiver.bal]: would be better I guess since these two follows REST API design best practices.

**Comment by manuranga** [comment]
We can't use `208` code for this.
`200` or `204` is appropriate for already created case.
`201` is more appropriate for newly created case.

File: idempotent-receiver/idempotent-receiver.bal [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+enum OrderStatus {
+    CREATED,
+    SHIPPED,
+    COMPLETED,
+    CANCELLED
+};
+
+map<OrderStatus> orderStatuses = {};
```
Comment (LakshanWeerasinghe) [inline idempotent-receiver/idempotent-receiver.bal]:
```suggestion
final map<OrderStatus> orderStatuses = {};
```

File: idempotent-receiver/idempotent-receiver.bal [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    CANCELLED
+};
+
+final map<OrderStatus> orderStatuses = {};
+
+service /api/v1 on new http:Listener(8080) {
+
+    resource function put manage\-orders/[string orderId](OrderDetail orderDetail) returns
+        http:STATUS_NO_CONTENT|http:STATUS_CREATED {
+            io:println(orderStatuses);
```
Comment (manuranga) [inline idempotent-receiver/idempotent-receiver.bal]: need to remove

**Review by manuranga — APPROVED** [review]

**Pull request merged**